frontend-react/src/components/Table/TableFilters.tsx

```diff
@@ -54,6 +52,17 @@ function TableFilters({ filterManager, cursorManager }: SubmissionFilterProps) {
         try {
             const from = new Date(rangeFrom).toISOString();
             const to = new Date(inclusiveDateString(rangeTo)).toISOString();
+
+            const eventName = featureEvent;
+            const eventData = {
+                tableFilter: {
+                    startRange: from,
+                    endRange: to,
+                },
+            };
+
+            TrackAppInsightEvent(eventName, eventData);
```

Comment:
Since TableFilters seems intended to be a generic component that supposedly could be rendered anywhere (_possibly_ without wanting to track analytics), it might be better to turn this into an `onFilterClick` prop:

```jsx
function TableFilters({ ...otherStuff, onFilterClick }) {
    // other stuff

    const applyToFilterManager = () => {
        updateRange();

        // call onFilterClick with the specified range
        onFilterClick({ from, to })
    };
}
```

Then we specify the tracking event in the parent components wherever we want to track:
```jsx
// in DeliveriesTable.tsx
<TableFilters
    ...otherStuff
    onFilterClick={ ({ from, to }) => TrackAppInsightEventName(FeatureType.DAILY_DATA, { tableFilter: { startRange: from, endRange: to } }) }
/>

// in SubmissionTable.tsx
<TableFilters
    ...otherStuff
    onFilterClick={ ({ from, to }) => TrackAppInsightEventName(FeatureType.SUBMISSIONS, { tableFilter: { startRange: from, endRange: to } }) }
/>
```

If we wanted to consolidate this into a utility function, we could also do that:

```jsx
function trackFilterClickEvent(eventName, { from, to })) {
    TrackAppInsightEventName(eventName, { tableFilter: { startRange: from, endRange: to } });
}
```

It's a small thing right now since TableFilters isn't used a lot of places, but this sort of abstraction would help to keep the generic TableFilters less coupled to specific use cases and more flexible.

Reply:
Sounds good!